pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -228,6 +228,26 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
 				continue
 			}
 
+			// Check whether container is present in state, there may be 3 reasons why it's not present:
+			// - container is in Be/Bu pod
+			// - kubelet just restarted - state is empty
+			// - container has been removed from state by RemoveContainer call (DeletionTimestamp is set)
+			if _, ok := m.state.GetCPUSet(containerID); !ok {
+				if status.Phase == v1.PodRunning && pod.DeletionTimestamp == nil {
+
+					glog.Infof("[cpumanager] reconcileState: container is not present in state - trying to add (pod: %s, container: %s, container id: %s)", pod.Name, container.Name, containerID)
+					err := m.AddContainer(pod, &container, containerID)
```

Comment:
@flyingcougar I don't think this is the right approach to handle Kubelet restarts for Gu pods. `AddContainer` may assign different set of CPUs than what was previously assigned via `allocateCPUs`->`takeByTopology`. If the container is running we should inspect it's cpuset and assign the previously assigned CPUs directly using `SetCPUSet` or some other mechanism. 

@ConnorDoyle WDYT?

Reply:
Agree. The cpuset may be different than it was before the kubelet restart. The "size" of cpuset will remain consistent. With i.e cgroup inspection we can retain the original cpuset.